### PR TITLE
refactor(download): Rename `dl()` to `Invoke-Download()`

### LINKED ISSUE #3830: Do not use `dl` as a name for function
`dl` is terrible function name for downloading files scoop that can leak into profile alias

### PR DESCRIPTION
**Pull request by examosa**
#### Description

This PR renames the `dl` and related functions to use more descriptive names that better adhere to PowerShell conventions.

#### Motivation and Context

Resolves #3830

#### How Has This Been Tested?

Test suite run on local machine and in [CI](https://github.com/examosa/Scoop/actions/runs/3040324142/jobs/4896244634); all remain passing.

#### Checklist:

- [x] I have read the [Contributing Guide](https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md).
- [x] I have ensured that I am targeting the `develop` branch.
- [x] I have updated the documentation accordingly.
- [x] I have updated the tests accordingly.
- [x] I have added an entry in the CHANGELOG.

### CODE REVIEW

**Comment by rashil2000** [comment]
I think `Invoke-Download` is good enough (`web` is implied)

File: bin/install.ps1 [hunk 1]
```diff
@@ -41,10 +41,10 @@ if (Get-Command -Name 'scoop' -ErrorAction SilentlyContinue) {
 $dir = ensure (versiondir 'scoop' 'current')
 
 # download scoop zip
-$zipurl = 'https://github.com/ScoopInstaller/Scoop/archive/master.zip'
-$zipfile = "$dir\scoop.zip"
+$zipUrl = "https://github.com/ScoopInstaller/Scoop/archive/master.zip"
```
Comment (niheaven) [inline bin/install.ps1:44]:
```suggestion
$zipUrl = 'https://github.com/ScoopInstaller/Scoop/archive/master.zip'
```

File: bin/install.ps1 [hunk 2]
```diff
@@ -57,11 +57,11 @@ shim "$dir\bin\scoop.ps1" $false
 
 # download main bucket
 $dir = "$scoopdir\buckets\main"
-$zipurl = 'https://github.com/ScoopInstaller/Main/archive/master.zip'
-$zipfile = "$dir\main-bucket.zip"
-Write-Output 'Downloading main bucket...'
-New-Item $dir -Type Directory -Force | Out-Null
-dl $zipurl $zipfile
+$zipUrl = "https://github.com/ScoopInstaller/Main/archive/master.zip"
```
Comment (niheaven) [inline bin/install.ps1:60]:
```suggestion
$zipUrl = 'https://github.com/ScoopInstaller/Main/archive/master.zip'
```

File: lib/install.ps1 [hunk 3]
```diff
@@ -49,7 +49,7 @@ function install_app($app, $architecture, $global, $suggested, $use_cache = $tru
     $original_dir = $dir # keep reference to real (not linked) directory
     $persist_dir = persistdir $app $global
 
-    $fname = dl_urls $app $version $manifest $bucket $architecture $dir $use_cache $check_hash
+    $fname = Import-Url $app $version $manifest $bucket $architecture $dir $use_cache $check_hash
```
Comment (niheaven) [inline lib/install.ps1:52]:
```suggestion
    $fname = Invoke-ScoopDownload $app $version $manifest $bucket $architecture $dir $use_cache $check_hash
```

**Review by niheaven — CHANGES_REQUESTED** [review]

**Comment by examosa** [comment]
Thanks for the feedback @niheaven. I have a few clarifying questions:

1. It looks like you're suggesting to remove `Invoke-WebDownload` and use `Invoke-WebRequest` directly. Will it be alright to omit the headers from the original code and the `-UseBasicParsing` flag ([PowerShell 5 compatibility](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/invoke-webrequest?view=powershell-5.1#:~:text=By%20default%2C%20script%20code%20in%20the%20web%20page%20may%20be%20run%20when%20the%20page%20is%20being%20parsed%20to%20populate%20the%20ParsedHtml%20property.%20Use%20the%20%2DUseBasicParsing%20switch%20to%20suppress%20this.))?
2. Based on the suggested changes to string quotes, would it be alright if I updated the VS Code workspace settings like so?

```patch
@@ -4,6 +4,7 @@
     "powershell.codeFormatting.preset": "OTBS",
     "powershell.codeFormatting.alignPropertyValuePairs": true,
     "powershell.codeFormatting.ignoreOneLineBlock": true,
+    "powershell.codeFormatting.useConstantStrings": true,
     "files.exclude": {
         "**/.git": true,
         "**/.svn": true,
```

**Comment by niheaven** [comment]
> 1. It looks like you're suggesting to remove `Invoke-WebDownload` and use `Invoke-WebRequest` directly. Will it be alright to omit the headers from the original code and the `-UseBasicParsing` flag ([PowerShell 5 compatibility](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/invoke-webrequest?view=powershell-5.1#:~:text=By%20default%2C%20script%20code%20in%20the%20web%20page%20may%20be%20run%20when%20the%20page%20is%20being%20parsed%20to%20populate%20the%20ParsedHtml%20property.%20Use%20the%20%2DUseBasicParsing%20switch%20to%20suppress%20this.))?

Yes, it should, but in fact this file is no longer used. Now installer script is [here](https://github.com/ScoopInstaller/Install/blob/master/install.ps1) in https://github.com/ScoopInstaller/Install.

> 2. Based on the suggested changes to string quotes, would it be alright if I updated the VS Code workspace settings like so?
> 
> ```diff
> @@ -4,6 +4,7 @@
>      "powershell.codeFormatting.preset": "OTBS",
>      "powershell.codeFormatting.alignPropertyValuePairs": true,
>      "powershell.codeFormatting.ignoreOneLineBlock": true,
> +    "powershell.codeFormatting.useConstantStrings": true,
>      "files.exclude": {
>          "**/.git": true,
>          "**/.svn": true,
> ```

Yes, it is, and we usually use the following setting too:

```json
"powershell.codeFormatting.useCorrectCasing": true
```

**Review by niheaven — APPROVED** [review]

**Comment by rashil2000** [comment]
@examosa Can you make a corresponding PR to [GithubActions](https://github.com/ScoopInstaller/GithubActions) too? Otherwise the actions will break when this feature gets released.

**Pull request merged**